src/kibana/components/state_management/app_state.js

```diff
@@ -26,13 +30,19 @@ define(function (require) {
   .factory('AppState', function (Private) {
     return Private(AppStateProvider);
   })
-  .service('getAppState', function () {
+  .service('getAppState', function ($location) {
     var currentAppState;
 
     function get() {
       return currentAppState;
     }
 
+    // Checks to see if the appState might already exist, even if it hasn't been newed up
+    get.mightExist = function () {
```

Comment:
Can you rename this to exists, or isLoaded or something? The might seems ambiguous, and I kind of expect it to take some action instead of just returning a bool.